File: for-local-test/226.invert-binary-tree.java

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        TreeNode temp = root.left;      
        root.left = invertTree(root.right);  
        root.right = invertTree(temp);
        return root;
    }
    public void printTree(TreeNode node){
        if(node == null) return;
        System.out.print(node.val + " ");
        printTree(node.left);
        printTree(node.right);
    }
    public static void main(String[] args){
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);
        Solution solution = new Solution();
        solution.printTree(root); // Output: 1 2 4 5 3
        System.out.println();
        TreeNode invertedRoot = solution.invertTree(root);
        
        System.out.print("Inverted tree (pre-order): ");
        solution.printTree(invertedRoot);
        System.out.println();
    }
}

File: for-local-test/python/226.invert-binary-tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root == None:
            return None
        temp = root.left
        root.left = self.invertTree(root.right)
        root.right = self.invertTree(temp)
        return root
    def printTree(self, root: TreeNode) -> None:
        if root == None:
            return
        print(root.val)
        self.printTree(root.left)
        self.printTree(root.right)
if __name__ == "__main__":
    solution = Solution()
    root = TreeNode(1, TreeNode(2, TreeNode(4), TreeNode(5)), TreeNode(3))
    print("Original Tree:")
    solution.printTree(root)
    inverted_root = solution.invertTree(root)
    print("Inverted Tree:")
    solution.printTree(inverted_root)
